docs/UsingOurScripts.md

```diff
@@ -0,0 +1,122 @@
+        // Weekly job template (e.g. weekly-openjdk8-pipeline)
+        "weekly"             : "pipelines/jobs/weekly_release_pipeline_job_template.groovy"
+    },
+    // Job configuration file paths (relative to this repository root)
+    "configDirectories"      : {
+        // Build configs directory containing node details, os, arch, testlists, etc
+        "build"              : "pipelines/jobs/configurations",
+        // Nightly configs directory containing execution frequency, weekly tags, platforms to build.
+        "nightly"            : "pipelines/jobs/configurations",
+        // Bash platform script directory containing jdk downloading and toolchain setups.
+        "platform"           : "build-farm/platform-specific-configurations"
+    },
+    // Job script paths (relative to this repository root)
+    "scriptDirectories"      : {
+        // Upstream scripts directory containing the 1st files that are executed by the openjdkx-pipeline jobs.
+        "upstream"           : "pipelines/build",
+        // Upstream script file containing the 1st script that is executed by the weekly-openjdk8-pipeline jobs.
+        "weekly"             : "pipelines/build/common/weekly_release_pipeline.groovy",
+        // Downstream script file containing the 1st script that is executed by the jdkx-platform-arch-variant jobs.
+        "downstream"         : "pipelines/build/common/kick_off_build.groovy",
+        // Base script file containing the 2nd script that is executed by the pipeline_jobs_generator_jdkxx jobs
+        "regeneration"       : "pipelines/build/common/config_regeneration.groovy",
+        // Base PR tester file script file containing the 2nd script that is executed by the pipeline_jobs_generator_jdkxx jobs
+        "tester"             : "pipelines/build/prTester/pr_test_pipeline.groovy"
+    },
+    // Job base file (the main file which is called after the 1st setup script file) paths (relative to this repository root)
+    "baseFileDirectories": {
+        // Upstream pipeline file script containing the 2nd script that is executed by the openjdkx-pipeline jobs
+        "upstream"           : "pipelines/build/common/build_base_file.groovy",
+        // Upstream pipeline file script containing the 2nd script that is executed by the jdkx-platform-arch-variant jobs
+        "downstream"         : "pipelines/build/common/openjdk_build_pipeline.groovy"
+    },
+    // Script to import the adopt groovy class library (relative to this repository root)
+    "importLibraryScript"    : "pipelines/build/common/import_lib.groovy"
+}
+```
+
+### How do I know which parameter the jenkins job will use?
+
+The scripts have been designed with a set hierarchy in mind when choosing which parameter to use:
+
+```md
+1. JENKINS PARAMETERS (highest priority, args entered here will be what the build scripts use over everything else)
+2. USER JSON (medium priority, args entered here will be pulled in and used when a jenkins parameter isn't entered)
```

Comment:
Wouldn't User override Adopt defaults?

Reply:
Correct. The document is stating that user jenkins parameters come first, then user default parameters (based on the `defaults.json` file in the user's repository) and then Adopt's default parameters last (based on the `defaults.json` file in this repository)

Reply:
Sorry I was unclear!  I figured the order (subtle change, but I think it makes more sense to the reader and how the logic should be structured?) would be:

1. JENKINS PARAMETERS (highest priority, args entered here will be what the build scripts use over everything else)
2. ADOPT JSON
3. USER JSON (medium priority, args entered here will be pulled in and used when a jenkins parameter isn't entered)